src/sheets.rs

```diff
@@ -35,12 +35,26 @@ pub(crate) fn cell_date(cell: &CellData) -> Result<chrono::NaiveDate, anyhow::Er
 pub(crate) async fn sheets_client(
     session: &Session,
     server_state: ServerState,
+    headers: HeaderMap,
     original_uri: Uri,
 ) -> Result<SheetsClient, Error> {
-    let maybe_token: Option<String> = session
-        .get(GoogleScope::Sheets.token_session_key())
-        .await
-        .context("Session load error")?;
+    let maybe_token = if let Some(auth_header) = headers.get("x-authorization-google") {
+        let token = match auth_header.to_str() {
+            Ok(s) => Some(s.to_string()),
+            Err(e) => {
+                return Err(Error::UserFacing(format!(
+                    "Invalid token in the header: {}",
```

Comment:
Let's include the header name in the error message (and extract a `const` for the header name so we can share it across the two uses) - otherwise LGTM, thanks!